test/hotspot/jtreg/serviceability/dcmd/thread/ClassInitMonitorVThread.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jdk.test.lib.dcmd.PidJcmdExecutor;
+import jdk.test.lib.process.OutputAnalyzer;
+import java.util.concurrent.CountDownLatch;
+
+/*
+ * @test
+ * @bug 8356222
+ * @summary Test jcmd Thread.print command for "waiting on the Class initialization monitor" case
+ * @requires vm.continuations
+ * @library /test/lib
+ * @run main ClassInitMonitorVThread
+ */
+
+class LongInitClass {
+    static {
+        ClassInitMonitorVThread.longInitClass_waiting = true;
+        while (ClassInitMonitorVThread.longInitClass_wait) {
+            try {
+                Thread.sleep(10);
+            } catch (Exception ex) {
+            }
+        }
+        ClassInitMonitorVThread.longInitClass_waiting = false;
+    }
+
+    LongInitClass() {}
+}
+
+public class ClassInitMonitorVThread {
+    static volatile boolean longInitClass_wait;
+    static volatile boolean longInitClass_waiting;
+
+    public static void main(String[] args) throws InterruptedException {
+        try {
+            // 1st thread starts class initialization
+            longInitClass_wait = true;
+            longInitClass_waiting = false;
+            Thread vthread1 = Thread.ofVirtual().name("Loader1").start(new Loader(null));
+            while (!longInitClass_waiting) {
+                Thread.sleep(10);
+            }
+
+            // 2nd thread is blocked at class initialization
+            // thread state is "RUNNING", so just wait some time after the thread is ready
+            CountDownLatch loaderReady = new CountDownLatch(1);
+            Thread vthread2 = Thread.ofVirtual().name("Loader2").start(new Loader(loaderReady));
+            loaderReady.await();
+            Thread.sleep(100);
```

Comment:
How reliable is this? With a debug build and -Xcomp/other options, then this might not long enough to observe "Loader 2" waiting for the class initializer.